cryptofeed/callback.py

```diff
@@ -6,102 +6,70 @@
 '''
 import asyncio
 import inspect
-from decimal import Decimal
 
 
 class Callback:
     def __init__(self, callback):
         self.callback = callback
         self.is_async = inspect.iscoroutinefunction(callback)
 
-    async def __call__(self, *args, **kwargs):
+    async def __call__(self, obj, receipt_timestamp, **kwargs):
         if self.callback is None:
             return
         elif self.is_async:
-            await self.callback(*args, **kwargs)
+            await self.callback(obj, receipt_timestamp, **kwargs)
         else:
             loop = asyncio.get_event_loop()
-            await loop.run_in_executor(None, self.callback, *args, **kwargs)
+            await loop.run_in_executor(None, self.callback, (obj, receipt_timestamp, kwargs))
 
 
 class TradeCallback(Callback):
-
-    def __init__(self, callback, include_order_type=False):
-        """
-        include_order_type is currently supported only on Kraken and Coinbase and enables
-        the order_type field in callbacks, which contains information about the order type (market/limit).
-
-        Note that to receive order_type on Coinbase, you must also subscribe to the L3_BOOK channel (though
-        do not need to specify any L3_BOOK callbacks)
-        """
-        self.include_order_type = include_order_type
-        super().__init__(callback)
-
-    async def __call__(self, *, feed: str, symbol: str, side: str, amount: Decimal, price: Decimal, order_id=None, timestamp: float, receipt_timestamp: float, order_type: str = None):
-        kwargs = {}
-        if self.include_order_type:
-            kwargs['order_type'] = order_type
-        await super().__call__(feed, symbol, order_id, timestamp, side, amount, price, receipt_timestamp, **kwargs)
+    pass
 
 
 class TickerCallback(Callback):
-    async def __call__(self, *, feed: str, symbol: str, bid: Decimal, ask: Decimal, timestamp: float, receipt_timestamp: float, **kwargs):
-        await super().__call__(feed, symbol, bid, ask, timestamp, receipt_timestamp, **kwargs)
+    pass
 
 
 class BookCallback(Callback):
-    """
-    For full L2/L3 book updates
-    """
-    async def __call__(self, *, feed: str, symbol: str, book: dict, timestamp: float, receipt_timestamp: float, **kwargs):
-        await super().__call__(feed, symbol, book, timestamp, receipt_timestamp, **kwargs)
-
-
-class BookUpdateCallback(Callback):
-    """
-    For Book Deltas
-    """
-    async def __call__(self, *, feed: str, symbol: str, delta: dict, timestamp: float, receipt_timestamp: float, **kwargs):
-        """
-        Delta is in format of:
-        {
-            BID: [(price, size), (price, size), ...]
-            ASK: [(price, size), (price, size), ...]
-        }
-        prices with size 0 should be deleted from the book
-        """
-        await super().__call__(feed, symbol, delta, timestamp, receipt_timestamp, **kwargs)
```

Comment:
This change is going to affect our market data repository. I think we have to move to submitting book delta updates on the same callback as book updates. Can you see any issues with that?

The function signatures will also change but that doesn't feel like that big of a deal.